From 0ba819433fd3f47c0285e752701752f615d57cb6 Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf.samuelsson@atmel.com>
Date: Sat, 16 Apr 2011 01:30:13 +0200
Subject: [PATCH 10/24] net: macb: modify SRAM location of descriptor buffers for 9xe chips

At91sam9xe chips have internal FLASH instead of SRAM at first address. We can't
locate macb descriptor buffers in this location. Instead put them in the
alternative location.

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>

Ported to linux-2.6.38
Signed-off-by: Ulf Samuelsson <ulf.samuelsson@atmel.com>
---
 drivers/net/macb.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/macb.c b/drivers/net/macb.c
index 16a3284..33f5eef 100644
--- a/drivers/net/macb.c
+++ b/drivers/net/macb.c
@@ -748,7 +748,8 @@ static int macb_alloc_consistent(struct macb *bp)
 
 #if defined(CONFIG_ARCH_AT91) && defined(CONFIG_MACB_TX_SRAM)
 #if  defined(CONFIG_ARCH_AT91SAM9260)
-	if (request_mem_region(AT91SAM9260_SRAM0_BASE, TX_DMA_SIZE, "macb")) {
+	if (!cpu_is_at91sam9xe()
+	&& request_mem_region(AT91SAM9260_SRAM0_BASE, TX_DMA_SIZE, "macb")) {
 		bp->tx_ring_dma = AT91SAM9260_SRAM0_BASE;
 	} else {
 		if (request_mem_region(AT91SAM9260_SRAM1_BASE, TX_DMA_SIZE, "macb")) {
-- 
1.6.3.3

